src/python/pants/util/dirutil.py

```diff
@@ -19,6 +19,23 @@
 from pants.util.strutil import ensure_text
 
 
+def longest_dir_prefix(path, prefixes):
```

Comment:
It might (or might not) be more efficient to use a trie (which would have to be refactored out of the source roots code), but I don't mind if that happens in a followup change, and only if there's a noticeable performance gain. The SourceRootTrie code splits the paths, generating small temporary strings that then need to be GC'd, whereas I assume `str.startswith` is highly optimized in the python interpreter, so this simple implementation might actually be faster in practice, for small numbers of prefixes...